examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -377,7 +376,7 @@ bool LockManager::GetUser(chip::EndpointId endpointId, uint16_t userIndex, Ember
     uint16_t credentialSize = static_cast<uint16_t>(sizeof(CredentialStruct) * userInStorage.currentCredentialCount);
```

Comment:
If `userInStorage.currentCredentialCount` is larger than kMaxCredentialsPerUser for some reason (e.g. firmware update changed the max value), you get a buffer overflow here.   Of course in that situation you have other problems too, like not being able to load all the credentials....  But it's bad if some random bit-flip somewhere in storage leads to out of bounds writes in code.

Reply:
Good call. Added a check for userInStorage.currentCredentialCount <= kMaxCredentialsPerUser to prevent a buffer overflow in this case